ykrt/ykrt.h

```diff
@@ -0,0 +1,55 @@
+#include <stdint.h>
+
+/// A meta-tracer.
+struct MT;
+typedef struct MT MT;
+
+// A `Location` stores state that the meta-tracer needs to identify hot loops
+// and run associated machine code.
+//
+// Each position in the end user's program that may be a control point (i.e.
+// the possible start of a trace) must have an associated `Location`. The
+// `Location` does not need to be at a stable address in memory and can be
+// freely moved.
+//
+// Program positions that can't be control points don't need an associated
+// `Location`. For interpreters that can't (or don't want) to be as selective,
+// a simple (if moderately wasteful) mechanism is for every bytecode or AST
+// node to have its own `Location` (even for bytecodes or nodes that can't be
+// control points).
+typedef struct {
+    uintptr_t state;
+} Location;
+
+#if defined(__x86_64)
```

Comment:
Would it be better to do the test based on pointer size rather than platform?

Reply:
I couldn't work out how to do that portably :/ This was my hack on the basis that it fails fast: if `__X86_64` isn't defined you get a compile-time error.

Reply:
You're right: https://scaryreasoner.wordpress.com/2009/02/28/checking-sizeof-at-compile-time/

I guess we leave it as-is then.